Source/WebInspectorUI/UserInterface/Views/EventBreakpointPopover.js

```diff
@@ -69,94 +59,104 @@ WI.EventBreakpointPopover = class EventBreakpointPopover extends WI.BreakpointPo
+        domEventNameEditorElement.classList.add("editor");
+
+        this._domEventNameCodeMirror = WI.CodeMirrorEditor.create(domEventNameEditorElement, {
+            extraKeys: {"Tab": false, "Shift-Tab": false},
+            lineWrapping: false,
+            mode: "text/plain",
+            matchBrackets: true,
+            scrollbarStyle: null,
+        });
+
+        this._domEventNameCompletionController = new WI.CodeMirrorCompletionController(WI.CodeMirrorCompletionController.Mode.Basic, this._domEventNameCodeMirror, this);
+
+        this._domEventNameCodeMirror.addKeyMap({
+            "Enter": () => {
+                this._domEventNameCompletionController.commitCurrentCompletion();
+                this.dismiss();
+            },
+            "Shift-Enter": () => {
+                this._domEventNameCompletionController.commitCurrentCompletion();
+                this.dismiss();
+            },
+            "Esc": () => {
+                this.dismiss();
+            },
+        });
+
+        let domEventNameInputElement = this._domEventNameCodeMirror.getInputField();
+        domEventNameInputElement.id = "edit-breakpoint-popover-content-event-name";
+
+        eventLabelElement.setAttribute("for", domEventNameInputElement.id);
+
+        if (WI.EventBreakpoint.supportsCaseSensitive) {
+            let caseSensitiveLabel = content.appendChild(document.createElement("label"));
+            caseSensitiveLabel.className = "case-sensitive";
+
+            this._caseSensitiveCheckboxElement = caseSensitiveLabel.appendChild(document.createElement("input"));
+            this._caseSensitiveCheckboxElement.type = "checkbox";
+            this._caseSensitiveCheckboxElement.checked = true;
+
+            caseSensitiveLabel.append(WI.UIString("Case Sensitive"));
+        }
 
-        eventLabelElement.setAttribute("for", this._domEventNameInputElement.id);
+        if (WI.EventBreakpoint.supportsIsRegex) {
+            let isRegexLabel = content.appendChild(document.createElement("label"));
+            isRegexLabel.className = "is-regex";
 
-        this.addRow("event", eventLabelElement, this._domEventNameInputElement);
+            this._isRegexCheckboxElement = isRegexLabel.appendChild(document.createElement("input"));
+            this._isRegexCheckboxElement.type = "checkbox";
+            this._isRegexCheckboxElement.checked = false;
+            this._isRegexCheckboxElement.addEventListener("change", (event) => {
+                this._updateDOMEventNameCodeMirrorMode();
+            });
+            this._updateDOMEventNameCodeMirrorMode();
+
+            isRegexLabel.append(WI.UIString("Regular Expression"));
+        }
+
+        this.addRow("event", eventLabelElement, content);
 
         // Focus the event name input after the popover is shown.
         setTimeout(() => {
-            this._domEventNameInputElement.focus();
+            this._domEventNameCodeMirror.refresh();
```

Comment:
Aside: Shame we don't have a `didPresentPopover` delegate call from the popover...

Reply:
heh sadly even if we did, we'd probably still need a `setTimeout`/`requestAnimationFrame` because `CodeMirror` initialization is not (always) synchronous